admin/tools/provision/provision-qserv.py

```diff
@@ -0,0 +1,140 @@
+        # overlay and selinux are not compliant in docker 1.9
+        - [ sed, -i, -e, "s/OPTIONS='--selinux-enabled'/# OPTIONS='--selinux-enabled'/", /etc/sysconfig/docker ]
+        - [ /bin/systemctl, daemon-reload]
+        - [ /bin/systemctl, restart,  docker.service]
+        '''
+    fpubkey = open(os.path.expanduser(cloudManager.key_filename + ".pub"))
+    public_key=fpubkey.read()
+    userdata = cloud_config_tpl.format(key=public_key)
+
+    return userdata
+
+
+if __name__ == "__main__":
+    try:
+        # Define command-line arguments
+        parser = argparse.ArgumentParser(description='Boot instances from image containing Docker.')
+        parser.add_argument('-n', '--nb-servers', dest='nbServers',
+                           required=False, default=3, type=int,
+                           help='Choose the number of servers to boot')
+
+        cloudmanager.add_parser_args(parser)
+        args = parser.parse_args()
+
+        loggerName = "Provisioner"
+        cloudmanager.config_logger(loggerName, args.verbose, args.verboseAll)
+
+        cloudManager = cloudmanager.CloudManager(config_file_name=args.configFile, used_image_key=cloudmanager.SNAPSHOT_IMAGE_KEY, add_ssh_key=True)
+
+        cloudManager.manage_ssh_key()
+
+        # Find a floating ip address for gateway
+        floating_ip = cloudManager.get_floating_ip()
+        if not floating_ip:
+            logging.fatal("Unable to add public ip to Qserv gateway")
+            sys.exit(1)
+
+        userdata_provision = get_cloudconfig()
+
+        # Create instances list
+        instances = []
+
+        # Create gateway instance and add floating_ip to it
+        gateway_id = 0
+        gateway_instance = cloudManager.nova_servers_create(gateway_id,
+                                                            userdata_provision)
+        logging.info("Add floating ip ({0}) to {1}".format(floating_ip,
+            gateway_instance.name))
+        gateway_instance.add_floating_ip(floating_ip)
+
+        # Manage ssh security group
+        if cloudManager.ssh_security_group:
+            gateway_instance.add_security_group(cloudManager.ssh_security_group)
+
+        instances.append(gateway_instance)
+
+        # Create worker instances
+        for instance_id in range(1, args.nbServers):
+            worker_instance = cloudManager.nova_servers_create(instance_id,
+                                                               userdata_provision)
+            instances.append(worker_instance)
+
+        cloudManager.print_ssh_config(instances, floating_ip)
+
+        # Wait for cloud config completion for all machines
+        for instance in instances:
+            cloudManager.detect_end_cloud_config(instance)
+
+        cloudManager.check_ssh_up(instances)
+
+        cloudManager.update_etc_hosts(instances)
+
+        logging.debug("SUCCESS: Qserv Openstack cluster is up")
+
+    except Exception as exc:
```

Comment:
`try` block is very long, I'd move its contents in a separate method (e.g. `main()`)